model-optimizer/install_prerequisites/install_prerequisites.sh

```diff
@@ -83,18 +83,117 @@ elif [[ "$OSTYPE" == "darwin"* ]]; then
     python3 -m pip install --upgrade pip
 fi
 
+install_latest_ov() {
+    if $2; then
+        sudo -E "$1" -m pip install openvino
+    else
+        "$1" -m pip install openvino
+    fi
+}
+
+install_ov() {
+    if $2; then
+        sudo -E "$1" -m pip install openvino=="$3"
+    else
+        "$1" -m pip install openvino=="$3"
+    fi
+}
+
+uninstall_ov() {
+    if $2; then
+        sudo -E "$1" -m pip uninstall -y openvino
+    else
+        "$1" -m pip uninstall -y openvino
+    fi
+}
+
+check_ie() {
+    $1 "$SCRIPTDIR/../mo/utils/find_ie_version.py"
+}
+
+check_ov_package() {
+    if $2; then
+        sudo -E "$1" -m pip show openvino
+    else
+        "$1" -m pip show openvino
+    fi
+}
+
+extract_release_from_version() {
+    version="$("$1" -c "import sys; import os; \
+                        sys.path.append(os.path.join(\"$SCRIPTDIR\", os.pardir)); \
+                        from mo.utils.version import extract_release_version; \
+                        print(\"{}.{}\".format(*extract_release_version()))")"
+}
+
+find_ie_bindings() {
+    python_executable="$1"
+    requires_sudo="$2"
+
+    if ! check_ie "$python_executable"; then
+        # check pip installed openvino package
+        if check_ov_package "$python_executable" "$requires_sudo"; then
+            echo "[ WARNING ] Existing OpenVINO version doesn't work as expected."
+            return 1
+        fi
+
+        # try to find OpenVINO version that matches MO version
+        if extract_release_from_version "$python_executable"; then
+            if [[ $version == "None.None" ]]; then
+                echo "[ WARNING ] Can not extract release version from ModelOptimizer version. The latest OpenVINO version will be installed (may be incompatible with current ModelOptimizer version)"
+            elif install_ov "$python_executable" "$requires_sudo" "$version"; then
+                if check_ie "$python_executable"; then
+                    return 0
+                fi
+                echo "[ WARNING ] Installed OpenVINO version doesn't work as expected...Uninstalling"
```

Comment:
Any diagnostics for the user?

Reply:
The detailed error message from failed import will be printed by python script.